### PR TITLE
Remove usage of deprecated OSAtomic methods

### PR DESCRIPTION
**Pull request by LumioseSil**
#### 7c820f37fe12b07ef482f5057898cad2ef5c650f
<pre>
Remove usage of deprecated OSAtomic methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=251865">https://bugs.webkit.org/show_bug.cgi?id=251865</a>

We should be using WTF/Atomics instead.

Reviewed by NOBODY (OOPS!).

*Source/WebCore/platform/ios/wak/WebCoreThread.mm: Removed libkern/OSAtomics import
*Source/WebKitLegacy/mac/WebView/WebView.mm: Made didDrawTiles a proper BOOL and replaced OSAtomicCompareAndSwap32 with atomicCompareExchangeStrong from WTF
*Source/WebKitLegacy/mac/WebView/WebViewData.h: Made didDrawTiles a proper BOOL
</pre>
https://github.com/WebKit/WebKit/commit/246bda184b8fff931db5db18cfb3d2e793deb825

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4898 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5023 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5195 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6414 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5012 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4891 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5197 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4995 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5255 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4983 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5061 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4385 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6431 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2549 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4381 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/9348 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4411 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4455 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/6055 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4874 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3964 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4374 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8439 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4735 "Built successfully") | | |

### CODE REVIEW

**Comment by AtariDreams** [comment]
@Constellation @weinig Good?

**Comment by AtariDreams** [comment]
@whsieh

File: Source/WebKitLegacy/mac/WebView/WebView.mm [hunk 1]
```diff
@@ -3474,7 +3472,7 @@ - (void)_setInitiatedDrag:(BOOL)initiatedDrag
 - (void)_didCommitLoadForFrame:(WebFrame *)frame
 {
     if (frame == [self mainFrame])
-        _private->didDrawTiles = 0;
+        _private->didDrawTiles = NO;
```
Comment (geoffreygaren) [inline Source/WebKitLegacy/mac/WebView/WebView.mm:3475]: This change does not appear to be related to OSAtomic.

File: Source/WebKitLegacy/mac/WebView/WebView.mm [hunk 2]
```diff
@@ -2312,10 +2312,8 @@ - (void)_dispatchTileDidDraw:(CALayer*)tile
         return;
     }
 
-    ALLOW_DEPRECATED_DECLARATIONS_BEGIN
-    if (!OSAtomicCompareAndSwap32(0, 1, &_private->didDrawTiles))
+    if (!WTF::atomicCompareExchangeWeakRelaxed(&_private->didDrawTiles, NO, YES))
```
Comment (geoffreygaren) [inline Source/WebKitLegacy/mac/WebView/WebView.mm]: These two functions are not equivalent.

**Review by geoffreygaren — CHANGES_REQUESTED** [review]

File: Source/WebKitLegacy/mac/WebView/WebView.mm (hunk 2 above)
Comment (AZero13) [reply Source/WebKitLegacy/mac/WebView/WebView.mm]: Examining the definitions of both functions, yes they are equivalent.

Comment (geoffreygaren) [reply Source/WebKitLegacy/mac/WebView/WebView.mm]: Can you post the definitions you examined, and define the architectures on which you think they're equivalent?

Comment (AZero13) [reply Source/WebKitLegacy/mac/WebView/WebView.mm]: Yes. Here 
Screen Shot 2023-02-21 at 8 53 29 AM
Screen Shot 2023-02-21 at 8 53 12 AM
Screen Shot 2023-02-21 at 8 51 26 AM

Comment (AZero13) [reply Source/WebKitLegacy/mac/WebView/WebView.mm]: I think strong is overkill, but if you deem it appropriate, I can fix that.

**Comment by AtariDreams** [comment]
@geoffreygaren thoughts?

**Comment by aproskuryakov** [comment]
What bug tracks this? https://bugs.webkit.org/show_bug.cgi?id=251865 is "reexport_install_name is no longer needed".

**Comment by AtariDreams** [comment]
> What bug tracks this? https://bugs.webkit.org/show_bug.cgi?id=251865 is "reexport_install_name is no longer needed".

My bad. Will fix.

**Comment by AtariDreams** [comment]
Fixed!

File: Source/WebKitLegacy/mac/WebView/WebView.mm (hunk 2 above)
Comment (geoffreygaren) [reply Source/WebKitLegacy/mac/WebView/WebView.mm]: When you say "I think strong is overkill", does that mean that you now see that these functions are not equivalent?

If so, can you explain the difference between these functions, and why you believe that difference is OK?

**Review by geoffreygaren — CHANGES_REQUESTED** [review]
My comment was marked as resolved, but it is not resolved.

File: Source/WebKitLegacy/mac/WebView/WebView.mm (hunk 2 above)
Comment (AZero13) [reply Source/WebKitLegacy/mac/WebView/WebView.mm]: Yes, one has a strong atomic compare which affects memory sequence, and the one I chose is more relaxed since the original code did not write OSAtomicCompare with the barrier.

Comment (geoffreygaren) [reply Source/WebKitLegacy/mac/WebView/WebView.mm]: No, that's not right. Please re-read the documentation on compare_exchange_weak.

**Comment by AtariDreams** [comment]
Done!

**Review by geoffreygaren — CHANGES_REQUESTED** [review]
My comment is still not resolved.

File: Source/WebKitLegacy/mac/WebView/WebView.mm (hunk 2 above)
Comment (AZero13) [reply Source/WebKitLegacy/mac/WebView/WebView.mm]: OK I understand it now. It should be strong.

**Comment by AtariDreams** [comment]
@geoffreygaren Is it resolved now?
